### ISSUE TITLE
Deleting gets stuck on the 100th comment

### ISSUE THREAD

**Issue by Matt07211** [Nick-Gottschlich/Social-Amnesia#37]
When ever I start the process to delete my comments it will get stuck on deleting the 100th comment.

**Comment by Nick-Gottschlich** [comment]
Thank you for reporting this! Was there any sort of error output, or does it just hang on the "processing item..."?

Additionally, are you able to interact with the app after it gets stuck? (This will help me determine if it's stuck in the deletion loop or crashed out of it).

One more thing, if you have time, it would help me if you could run the app from the command line (see here: https://github.com/Nick-Gottschlich/Social-Amnesia#how-to-run-social-amnesia-from-the-command-line) and check and see if the terminal is reporting any errors.

Thanks for trying out Social Amnesia!

**Comment by Matt07211** [comment]
Sorry I didn't include the needed info in the post when I submited it, I was rushing to post it before i needed to go somewhere. Now onto the useful information.

OS: Solus (Linux Distro)
Python: ```$ python3 --version Python 3.6.8```
Note: I am using 2fa on my reddit account if this changes anything at all

Now when I first encounted the problem, I had received this error in the terminal and the program just hanged (So I killed the program).
```bash
 $ python3 SocialAmnesia.py 
(<__main__.MainApp object .!mainapp>, <class 'prawcore.exceptions.OAuthException'>, OAuthException('invalid_grant error processing request',), <traceback object at 0x7efbf4b024c8>)
(<__main__.MainApp object .!mainapp>, <class 'prawcore.exceptions.OAuthException'>, OAuthException('invalid_grant error processing request',), <traceback object at 0x7efbf4b0a988>)
(<__main__.MainApp object .!mainapp>, <class 'prawcore.exceptions.OAuthException'>, OAuthException('invalid_grant error processing request',), <traceback object at 0x7efbf4b10908>)
(<__main__.MainApp object .!mainapp>, <class 'prawcore.exceptions.OAuthException'>, OAuthException('invalid_grant error processing request',), <traceback object at 0x7efbf4b15888>)
(<__main__.MainApp object .!mainapp>, <class 'prawcore.exceptions.OAuthException'>, OAuthException('invalid_grant error processing request',), <traceback object at 0x7efbf4af1d48>)
^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^CTraceback (most recent call last):
  File "SocialAmnesia.py", line 742, in <module>
    root.mainloop()
  File "/usr/lib/python3.6/tkinter/__init__.py", line 1283, in mainloop
    self.tk.mainloop(n)
KeyboardInterrupt
```

Second time I ran it the number of comments was below the 1K mark (I know i have more then a couple K comments, so i expected this to be back at the 1K line after running the app again and login in), I thought nothing of it at the time, ran the delete again, it got stuck at the 100th comment again, nothing in the terminal output, this time instead of forcing closing like I originally did i just clicked the delete button again, I repeated this about 3 times (I was doing other stuff in another terminal session and accidentaly killed the running of the program.

The 3rd time I ran the program I recived the same error again
```bash
$ python3 SocialAmnesia.py 
(<__main__.MainApp object .!mainapp>, <class 'prawcore.exceptions.OAuthException'>, OAuthException('invalid_grant error processing request',), <traceback object at 0x7efbf40ef508>)
^C^C^C^C^C^C^C^C^C^C^C^CTraceback (most recent call last):
  File "SocialAmnesia.py", line 742, in <module>
    root.mainloop()
  File "/usr/lib/python3.6/tkinter/__init__.py", line 1283, in mainloop
    self.tk.mainloop(n)
KeyboardInterrupt
``` 
So I killed it and left it as is (and quickly wrote this issue up) as I had other places to be by then.

Hopefully you can make use of the wall of text.

TL;DR ```OAuthException('invalid_grant error processing request',)``` and yes I can interact with the gui

**Comment by Nick-Gottschlich** [comment]
Ok, thank you for the detailed post. I can take a look at the `invalid_grant error processing request` and see what I come up with.

> Second time I ran it the number of comments was below the 1K mark (I know i have more then a couple K comments, so i expected this to be back at the 1K line after running the app again and login in)

So this I think I know, the reddit API only indexes the most recent 1,000 comments back, and I'm not sure how often (or if ever) it reindexes if comments are deleted. This is something I will need to test further, and probably add a warning in the actual app as opposed to just at the bottom of the readme [here](https://github.com/Nick-Gottschlich/Social-Amnesia#limitations).

**Comment by Nick-Gottschlich** [comment]
https://github.com/shadowmoose/RedditDownloader/issues/22#issuecomment-407703392

So I think this is a 2FA issue, and I think I will need to handle code flow (https://praw.readthedocs.io/en/latest/getting_started/authentication.html#code-flow) to make this work. For now, I've bumped the version of PRAW which may fix this (but I doubt it). I will enable 2FA on my tester account and dig into this.

**Comment by Nick-Gottschlich** [comment]
@Matt07211 I have implemented 2FA functionality for reddit, see this PR: https://github.com/Nick-Gottschlich/Social-Amnesia/pull/39
